### ISSUE TITLE
Yellow exclamation mark appears in VS Code

### ISSUE THREAD

**Issue by pcaversaccio** [juanfranblanco/vscode-solidity#257]
I'm really not sure whether this warning is triggered by this extension, however, I start here and any further help is appreciated. I get a yellow exclamation mark next to a low-level call. There is no further information when I hover over it nor any problems/warnings logged. Does someone have idea how to fix this (the code compiles also w/o any warnings): 
![image](https://user-images.githubusercontent.com/25297591/123057972-9e900e80-d408-11eb-973a-3d7be74a5065.png)
![image](https://user-images.githubusercontent.com/25297591/123058149-d008da00-d408-11eb-85cd-c3bdadcc1520.png)
UPDATE: If I comment the low-level call the exclamation mark disappears; thus it must be a connection to the Solidity code.

**Comment by cameel** [comment]
I wonder if it might be something in compiler's output that's triggering this. Can you provide a small example standalone contract that reproduces this?

**Comment by pcaversaccio** [comment]
Sure, let's take the [`MinimalForwarder.sol`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/metatx/MinimalForwarder.sol) from OpenZeppelin:
```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.0;

import "../utils/cryptography/ECDSA.sol";
import "../utils/cryptography/draft-EIP712.sol";

/*
 * @dev Simple minimal forwarder to be used together with an ERC2771 compatible contract. See {ERC2771Context}.
 */
contract MinimalForwarder is EIP712 {
    using ECDSA for bytes32;

    struct ForwardRequest {
        address from;
        address to;
        uint256 value;
        uint256 gas;
        uint256 nonce;
        bytes data;
    }

    bytes32 private constant TYPEHASH = keccak256("ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)");

    mapping(address => uint256) private _nonces;

    constructor() EIP712("MinimalForwarder", "0.0.1") {}

    function getNonce(address from) public view returns (uint256) {
        return _nonces[from];
    }

    function verify(ForwardRequest calldata req, bytes calldata signature) public view returns (bool) {
        address signer = _hashTypedDataV4(keccak256(abi.encode(
            TYPEHASH,
            req.from,
            req.to,
            req.value,
            req.gas,
            req.nonce,
            keccak256(req.data)
        ))).recover(signature);
        return _nonces[req.from] == req.nonce && signer == req.from;
    }

    function execute(ForwardRequest calldata req, bytes calldata signature) public payable returns (bool, bytes memory) {
        require(verify(req, signature), "MinimalForwarder: signature does not match request");
        _nonces[req.from] = req.nonce + 1;

        // solhint-disable-next-line avoid-low-level-calls
        (bool success, bytes memory returndata) = req.to.call{gas: req.gas, value: req.value}(abi.encodePacked(req.data, req.from));
        // Validate that the relayer has sent enough gas for the call.
        // See https://ronan.eth.link/blog/ethereum-gas-dangers/
        assert(gasleft() > req.gas / 63);

        return (success, returndata);
    }
}

```

The following line throws the warning:
![image](https://user-images.githubusercontent.com/25297591/123164421-3591b080-d473-11eb-9abf-c9c02254eb32.png)

**Comment by cameel** [comment]
Thanks. I don't see anything special in compiler's output (indeed no warnings or errors and the output itself seems correct) so this is likely something in the extension itself. Maybe something related to the suppressed solhint warning?

**Comment by pcaversaccio** [comment]
Ok good, this confirms also my findings. I already tested that - you can delete the solhint line but the warning stays. The only way to get rid of this specific warning is to comment the low level call (therefore it must somehow relate to this line...):
![image](https://user-images.githubusercontent.com/25297591/123168092-cb2f3f00-d477-11eb-8d24-c00bf96c013c.png)

**Comment by pcaversaccio** [comment]
Ok I figured out what is triggering this warning: it's the parameters in the curly brackets (in my case `{gas: req.gas}`). If you remove these arguments, everything disappears - @cameel could it be that the compiler throws something under the hood related to this?

**Comment by cameel** [comment]
Very unlikely. I wanted to try to see if it maybe getting a message with empty description or something but that's not the case. There are no messages at all in the output so the extension is not hiding anything. It must be something else.

**Comment by pcaversaccio** [comment]
Ok digged a little bit further; it seems that this [ConsenSys](https://github.com/ConsenSys/vscode-solidity-auditor) extension is triggering the warning (that's why I wrote it above that I'm not sure whether it's this extension). I will close this issue here and open a new issue in their repo.

**Comment by juanfranblanco** [comment]
@pcaversaccio glad you are sorted :) apologies I missed all this yesterday.

**Comment by pcaversaccio** [comment]
No worries!

**Comment by tintinweb** [comment]
cross-refing this here: https://github.com/ConsenSys/vscode-solidity-auditor/issues/78#issuecomment-867801024

TLDR; The [vscode visual developer](https://marketplace.visualstudio.com/items?itemName=tintinweb.solidity-visual-auditor) extension annotates lines that potentially call out to other contracts with a yellow ⚠️ icon in the gutter. This is to provide visual feedback about potential risk areas of the code. The feature can be disabled in the [settings](https://github.com/ConsenSys/vscode-solidity-auditor/issues/78#issuecomment-867801024). sorry for the confusion :)
